### ISSUE TITLE
Ttl parsing issue in az network dns record-set cname set-record 

### ISSUE THREAD

**Issue by mbv** [Azure/azure-cli#26248]
Unable to set cname record with ttl param. It worked on 2.47.0

**Command Name**
`az network dns record-set cname set-record`

**Errors:**
```
The command failed with an unexpected error. Here is the traceback:
Expect <class 'int'>, got 60 (<class 'str'>)
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/azure-cli/2.48.1/libexec/lib/python3.10/site-packages/knack/cli.py", line 233, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/homebrew/Cellar/azure-cli/2.48.1/libexec/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 663, in execute
    raise ex
  File "/opt/homebrew/Cellar/azure-cli/2.48.1/libexec/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/homebrew/Cellar/azure-cli/2.48.1/libexec/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 697, in _run_job
    result = cmd_copy(params)
  File "/opt/homebrew/Cellar/azure-cli/2.48.1/libexec/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 333, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/homebrew/Cellar/azure-cli/2.48.1/libexec/lib/python3.10/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
  File "/opt/homebrew/Cellar/azure-cli/2.48.1/libexec/lib/python3.10/site-packages/azure/cli/command_modules/network/custom.py", line 2833, in add_dns_cname_record
    return _add_save_record(cmd, record, record_type, record_set_name, resource_group_name, zone_name,
  File "/opt/homebrew/Cellar/azure-cli/2.48.1/libexec/lib/python3.10/site-packages/azure/cli/command_modules/network/custom.py", line 3093, in _add_save_record
    return _DNSRecordSetCreate(cli_ctx=cmd.cli_ctx)(command_args={
  File "/opt/homebrew/Cellar/azure-cli/2.48.1/libexec/lib/python3.10/site-packages/azure/cli/core/aaz/_command.py", line 149, in __call__
    return self._handler(*args, **kwargs)
  File "/opt/homebrew/Cellar/azure-cli/2.48.1/libexec/lib/python3.10/site-packages/azure/cli/command_modules/network/aaz/latest/network/dns/record_set/_create.py", line 26, in _handler
    super()._handler(command_args)
  File "/opt/homebrew/Cellar/azure-cli/2.48.1/libexec/lib/python3.10/site-packages/azure/cli/core/aaz/_command.py", line 153, in _handler
    self.ctx = AAZCommandCtx(
  File "/opt/homebrew/Cellar/azure-cli/2.48.1/libexec/lib/python3.10/site-packages/azure/cli/core/aaz/_command_ctx.py", line 32, in __init__
    self.args[dest] = cmd_arg
  File "/opt/homebrew/Cellar/azure-cli/2.48.1/libexec/lib/python3.10/site-packages/azure/cli/core/aaz/_field_value.py", line 76, in __setitem__
    self._data[name] = attr_schema.process_data(data, key=name)
  File "/opt/homebrew/Cellar/azure-cli/2.48.1/libexec/lib/python3.10/site-packages/azure/cli/core/aaz/_field_type.py", line 48, in process_data
    raise AAZInvalidValueError('Expect {}, got {} ({})'.format(self.DataType, data, type(data)))
azure.cli.core.aaz.exceptions.AAZInvalidValueError: Expect <class 'int'>, got 60 (<class 'str'>)
```

## To Reproduce:
use any number for ttl value, e.g. 60
- `az network dns record-set cname set-record --resource-group {} --zone-name {} --record-set-name {} --ttl {} --cname {}`

## Environment Summary
```
macOS-12.6.1-arm64-arm-64bit, Darwin 21.6.0
Python 3.10.11
Installer: HOMEBREW

azure-cli 2.48.1

Extensions:
azure-devops 0.26.0

Dependencies:
msal 1.20.0
azure-mgmt-resource 22.0.0
```

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by navba-MSFT** [comment]
@mbv Apologies for the late reply. Thanks for reaching out to us and reporting this issue. The above PR has been filed to address this issue and it is currently under review.

**Comment by navba-MSFT** [comment]
@mbv 2.49.0 The fix will be released around 05/23 in AzCLI 2.49.0. Workarounds can be found at https://github.com/Azure/azure-cli/issues/26274#issuecomment-1527066849.
